test/e2e/gameserver_test.go

```diff
@@ -395,20 +395,22 @@ func TestGameServerUnhealthyAfterReadyCrash(t *testing.T) {
 	}()
 	go func() {
 		for {
-			if atomic.LoadInt32(&stop) > 0 {
-				l.Info("UDP Crash stop signal received. Stopping.")
-				return
-			}
-			conn, err := net.Dial("udp", address)
-			assert.NoError(t, err)
-			defer conn.Close() // nolint: errcheck
-			_, err = conn.Write([]byte("CRASH"))
-			if err != nil {
-				l.WithError(err).Warn("error sending udp packet. Stopping.")
-				return
-			}
-			l.WithField("address", address).Info("sent UDP packet")
-			time.Sleep(5 * time.Second)
+			func() {
```

Comment:
Same reason as above. In this case, it makes more sense to release the conn that was created and used in the for loop at the end of each iteration. So wrap the code with a function to achieve it.

Reply:
The test failure that had a bunch of lines like: `time="2022-09-23 04:01:31.215" level=info msg="UDP Crash stop signal received. Stopping." test=TestGameServerUnhealthyAfterReadyCrash` was actually because in my previous commit I accidentally wrapped the `return` statement into the function so it only exit the wrapper function rather than the outsider for loop, which makes the loop infinite